### DISCUSSION TITLE
Zed Windows Build and Update Script

### DISCUSSION THREAD

**Discussion by da-moon** [zed-industries/zed#16811] [Show and tell]
# Zed Windows Build and Update Script

Hey everyone! 
I love this editor and want to give back to the community.

I wrote a PowerShell script that automates the building, installing, and updating Zed on Windows. 
With this script, you won't have to launch `zed` in WSL  which has it's own issues (e.g #14126)

Here are its key features:

## Automatic Update Checking

- Fetches the latest version information from GitHub
- Compares installed version with the latest available version
- Supports both stable and preview channels
- Only proceeds with build and installation if a newer version is available

## Flexible Installation

- Detects and supports multiple Windows architectures (x86_64, aarch64)
- Creates a version file for easy version tracking
- Installs Zed to a user-specific location

## Build Process

- Clones or updates the Zed repository
- Uses Rust with static linking for better portability
- Builds Zed using the appropriate target for the detected architecture

## Environment Setup

- Checks for and guides the installation of required dependencies (Rust, Visual Studio build tools, Windows SDK)
- Updates Rust to the latest version before building

## Command-line Integration

- Creates a PowerShell shim for running Zed from the command line
- Adds Zed to the user's PATH for easy access
- Allows running Zed with arguments (e.g., `zed --help`) from any terminal

## User Convenience

- Creates a desktop shortcut for quick access to Zed
- Provides clear console output about the installation and update process
- Runs diagnostics to ensure proper installation and functionality

## Channel Support

- Supports both stable and preview channels
- Uses the `ZED_CHANNEL` environment variable to determine which channel to use

## Version Management

- Implements a robust version comparison function to handle various version formats
- Stores the installed version for future update checks

## Error Handling and Diagnostics

- Implements error checking throughout the script
- Provides diagnostic information about the Zed executable and process

## Customization

- Allows for easy customization of installation directories and other parameters

This script provides a comprehensive solution for Windows users to easily install, update, and manage Zed, offering similar convenience to package managers on other platforms.

## Usage Guide

To use this script, follow these steps:

1. Download the script from the following gist:
   [https://gist.github.com/da-moon/bd9dd0f9f5164f06108727fa35bfe644](https://gist.github.com/da-moon/bd9dd0f9f5164f06108727fa35bfe644)

2. Save the script as `build-zed.ps1` on your local machine.

3. Open PowerShell as an administrator.

4. Navigate to the directory containing the script:
   ```powershell
   cd path\to\script\directory
   ```

5. (Optional) If you want to use the preview channel, set the environment variable:
   ```powershell
   $env:ZED_CHANNEL = "preview"
   ```

6. Run the script:
   ```powershell
   .\build-zed.ps1
   ```

7. Follow any on-screen prompts for installing dependencies if needed.

8. Once the script completes, you can run Zed using the `zed` command from any PowerShell or Command Prompt window.

### Using the Zed Command

After installation, you can use the `zed` command in the following ways:

- `zed`: Opens Zed in the current directory
- `zed .`: Opens Zed in the current directory (explicitly)
- `zed path/to/file`: Opens the specified file in Zed
- `zed --help`: Displays help information

> UPDATE: The improved shim ensures that Zed opens in the correct directory and doesn't block the PowerShell session when launching the GUI.

## Troubleshooting

If you encounter any issues while using the `zed` command, try the following:

1. Ensure that Zed is correctly installed by running the build script again.
2. Check if the Zed executable exists at the expected location (typically `$env:LOCALAPPDATA\Programs\Zed\zed.exe`).
3. If you receive an error about the file not being found, please ensure your PATH environment variable includes the Zed installation directory. The installer should have updated the `PATH`  ; if it did not work, try closing and opening a new Powershell terminal.
4. The shim will display any error messages for command-line operations (like `--help`). Read these carefully, as they may provide clues about what's going wrong.
5. If you're having trouble with the GUI not opening, try running `$env:LOCALAPPDATA\Programs\Zed\zed.exe` directly to see if there are any visible error messages.

If problems persist, please report the issue on the gist page, including any error messages and the steps to reproduce the problem.

> The script will automatically check for updates and only build/install if a newer version is available. You can run the script periodically to keep Zed up-to-date.

For any issues or contributions, please visit the gist page and leave a comment or create a fork with your improvements.

**Comment by da-moon** [comment]
# WSL SSH Setup Guide for Zed Remote Development

I was experimenting this weekend to get Zed remote development working on WSL2.
I'm excited to share a PowerShell script that simplifies the process of setting
up SSH access to your WSL2 environment, with specific enhancements for Zed
remote development. This script automates the installation and configuration of
an SSH server in WSL2, sets up port forwarding on Windows, and includes features
to improve the experience when using Zed for remote development.

## Prerequisites

- Make sure you have set wsl default user in `/etc/wsl.conf`

```
[user]
default = "<your username>"
```

## Gist

Here is the gist: https://gist.github.com/da-moon/51f23971feb005d5389642e56ce33bff

## Script Features

1. **Automatic WSL Distribution Detection**: The script attempts to detect your default WSL distribution or prompts you to specify one.

2. **SSH Server Setup in WSL2**: Installs and configures the SSH server in your chosen WSL2 distribution.

3. **Windows Firewall Configuration**: Sets up the necessary firewall rules to allow SSH access.

4. **Port Forwarding**: Configures port forwarding from Windows to WSL2 for SSH access.

5. **Persistent Configuration**: Creates a startup script to maintain port forwarding across reboots.

6. **SSH Key Generation**: Optionally generates an SSH key pair for secure authentication.

7. **Windows SSH Config Update**: Adds an entry to your SSH config file for easy access to your WSL2 environment.

8. **Zed-Specific Environment Variable Support**: Allows setting the `TERM_PROGRAM` and `COLORTERM` environment variables, which are useful for Zed remote development.

## Zed Environment Variable Support

A key feature of this script is its support for the `TERM_PROGRAM` and `COLORTERM` environment variables, which are particularly useful for Zed users. This functionality allows for customized configurations in WSL based on whether you're connecting through Zed.

## How to Use the Script

1. Download the `Setup-WSL2SSH.ps1` script from the provided gist.
2. Open PowerShell as Administrator.
3. Navigate to the directory containing the script.
4. Run the script:
   ```powershell
   .\Setup-WSL2SSH.ps1
   ```
5. Follow the on-screen prompts to complete the setup.

## Updating Zed Configuration

To ensure Zed sets the correct environment variables when using the integrated terminal on Windows, you can update the Zed configuration file. Note that this configuration is separate from the SSH configuration and only affects Zed's behavior on the Windows side.

Run the following PowerShell snippet to update the Zed configuration file:

```powershell
$jsonPath = "$env:APPDATA\Zed\settings.json"
$cleanedContent = (Get-Content -Path $jsonPath) | ForEach-Object {
    $_ -replace '//.*', ''
} | Out-String
$jsonContent = $cleanedContent | ConvertFrom-Json
if (-not $jsonContent.terminal) {
    $jsonContent | Add-Member -MemberType NoteProperty -Name "terminal" -Value @{ "env" = @{} }
} elseif (-not $jsonContent.terminal.env) {
    $jsonContent.terminal | Add-Member -MemberType NoteProperty -Name "env" -Value @{}
}
if (-not $jsonContent.terminal.env.PSObject.Properties["TERM_PROGRAM"]) {
    $jsonContent.terminal.env.TERM_PROGRAM = "zed"
}
$jsonContent | ConvertTo-Json -Depth 10 | Set-Content -Path $jsonPath -Force
```

**Note**: This snippet will remove any comments from your Zed configuration file. If you have important comments in your configuration, make sure to back up the file before running the snippet.

This snippet updates the Zed configuration to set the `TERM_PROGRAM` environment
variable for Zed's integrated terminal on **Windows**. Remember, this does not
affect the SSH connection to WSL directly; for that, you need to use the SSH
config file as described in the following section.

## Customizing Your WSL Environment for Zed

After running the setup script, you'll need to update your SSH configuration on Windows to take advantage of the Zed-specific features:

1. Open the SSH config file using Zed:
   ```powershell
   zed $env:USERPROFILE\.ssh\config
   ```

2. Add or modify the entry for your WSL connection to include the `SetEnv` options:

   ```
   Host wsl_your_distro_name
     # existing config ...
     SetEnv TERM_PROGRAM=zed
     SetEnv COLORTERM=truecolor
   ```



With the SSH configuration set up, you can customize your WSL environment based on whether you're connecting from Zed. For example, to prevent Zellij from automatically starting a session when using Zed's terminal, add this to your `.bashrc` or `.zshrc` in WSL:

```bash
if [ -x "$(command -v zellij)" ]; then
   terminal="$(ps -o comm= -p $(ps -o ppid= -p $$))"
   case "${terminal}" in
      "warp" | "tilda" | "poetry" | "node" | "zed-editor")
      # No action needed, terminal matches one of the conditions
      ;;
      *)
      if [ "${TERM_PROGRAM}" != "vscode" ] && [ "${TERM_PROGRAM}" != "zed" ]; then
         eval "$(zellij setup --generate-auto-start bash)"
      fi
   ;;
   esac
fi
```

## Troubleshooting

- If you encounter issues with SSH connection after a WSL restart, run the following command in PowerShell:
  ```powershell
  powershell $env:USERPROFILE\ConfigureWSLSSHForwarding.ps1
  ```
- Depending on your WSL/WSLg version, you might experience issues with your Wayland/X11 session. For potential solutions, refer to [this GitHub issue](https://github.com/microsoft/wslg/issues/1032#issuecomment-2310369848).

## Conclusion

This script aims to simplify the process of setting up SSH access to WSL2, with specific enhancements for Zed users. It provides a foundation for seamless remote development experiences. Feel free to customize the script further to suit your specific needs.

I welcome any feedback or suggestions for improvements. Happy coding with Zed and WSL!

**Comment by da-moon** [comment]
You can use the following snippet in your `$PROFILE` to automatically set Zed's LLM configuration environment variables from powershell password store:

```powershell
Import-Module Microsoft.PowerShell.SecretStore
# NOTE: use a safer password than `00000000` if needed !
Unlock-SecretStore -Password (ConvertTo-SecureString "00000000" -AsPlainText -Force)
# NOTE: initialize the `SecretStore` secret store
# Register-SecretVault -Name 'SecretStore' -ModuleName 'Microsoft.PowerShell.SecretStore'
# Set-SecretStorePassword -NewPassword (ConvertTo-SecureString "00000000" -AsPlainText -Force)
if (Get-Command Get-Secret -ErrorAction SilentlyContinue) {
  # NOTE: setting API key in the secret store
  # Set-Secret -Vault 'SecretStore' -Name 'OPENAI_API_KEY' -Secret "<YOUR-OPENAPI-TOKEN>"
  # NOTE: remove an element from the secret store
  # Remove-Secret -Vault 'SecretStore' -Name "OPENAI_API_KEY"
    if (Get-Secret -AsPlainText -Name 'OPENAI_API_KEY' -Vault 'SecretStore' -ErrorAction SilentlyContinue) {
        `$env:OPENAI_API_KEY = "$(Get-Secret -AsPlainText -Name 'OPENAI_API_KEY' -Vault 'SecretStore')"
    }
    if (Get-Secret -AsPlainText -Name 'ANTHROPIC_API_KEY' -Vault 'SecretStore' -ErrorAction SilentlyContinue) {
        `$env:ANTHROPIC_API_KEY = "$(Get-Secret -AsPlainText -Name 'ANTHROPIC_API_KEY' -Vault 'SecretStore')"
    }
    if (Get-Secret -AsPlainText -Name 'GOOGLE_AI_API_KEY' -Vault 'SecretStore' -ErrorAction SilentlyContinue) {
        `$env:GOOGLE_AI_API_KEY = "$(Get-Secret -AsPlainText -Name 'GOOGLE_AI_API_KEY' -Vault 'SecretStore')"
    }
}
```

Keep in mind that this approach won't pass in environment variables to Zed when it is launched through the start menu. To pass API keys to Zed , launched from start menu / shortcut, you have to persist it on User environment variables

```powershell
if (Get-Command Get-Secret -ErrorAction SilentlyContinue) {
    @('OPENAI_API_KEY', 'ANTHROPIC_API_KEY', 'GOOGLE_AI_API_KEY') | ForEach-Object {
        if (Get-Secret -AsPlainText -Name $_ -Vault 'SecretStore' -ErrorAction SilentlyContinue) {
            $secretValue = Get-Secret -AsPlainText -Name $_ -Vault 'SecretStore'
            [Environment]::SetEnvironmentVariable($_, $secretValue, 'User')
            Write-Host "$_ has been set from SecretStore and persisted at the user level."
        } else {
            Write-Host "$_ not found in SecretStore."
        }
    }
} else {
    Write-Host "SecretStore module not found or Get-Secret command not available."
}

Write-Host "Environment variables have been updated. You may need to restart your applications or log out and back in for the changes to take effect."
```

> Keep in mind that, in general, I would recommend against storing API keys on Windows. I don't trust the operating system, considering how easily it can be plagued by viruses.

**Comment by ablekh** [comment]
@da-moon Nice effort, much appreciated! I suggest you to add a "clean" or "uninstall" flag to enable a complete removal of Zed and, optionally, all or some of its dependencies (e.g., "-u All", "-u VSBuildTools", "-u WinSDK", "-u Rust") from the system.

**Reply by da-moon** [reply]
Installing dependencies is out of this script's scope;  It wasn't easy to install them unattended in Powershell. Feel free to submit your update to the gist in case you want to add an uninstall option.

**Reply by ablekh** [reply]
I agree. After some thinking, I also think that it might be better to keep installing / uninstalling dependencies out of scope for this script (even though it's a popular functionality, especially in Linux package managers). However, out of curiosity, I've done a brief research on the topic of unattended (aka silent) installation / removal of Microsoft's software using the tools' native CLI and/or PowerShell. In case you're curious about that as well, the following are some relevant resources.

https://silentinstallhq.com/windows-10-software-development-kit-sdk-silent-install-how-to-guide
https://silentinstallhq.com/windows-11-software-development-kit-sdk-silent-install-how-to-guide
https://silentinstallhq.com/visual-studio-build-tools-2022-silent-install-how-to-guide

Alternatively, there is Chocolatey (with which I have even more limited experience) and the following relevant packages.

https://community.chocolatey.org/packages/windows-sdk-10.1
https://community.chocolatey.org/packages/visualstudio2022buildtools
https://community.chocolatey.org/packages/rust (not sure if that's a full enough Rust distribution for building Zed)

**Reply by da-moon** [reply]
Yeah, you are right.
So, my thought process when I wrote this was that I wanted something to work on base Windows installs, meaning the install does not have a package manager like Chocolatey or Scoop (even Winget), and it runs `Powershell 5.1`.

This is because when I work with enterprise customers, this is the most common VDI config they give me. I can submit SNOW tickets and justifications for installing a dev tool like Visual Studio, but I wanted to keep dependencies minimal.

**Comment by Aashish-collectiv** [comment]
Hey @da-moon , the above script does not works for me

**Reply by da-moon** [reply]
Add `Set-PSDebug -Trace 2` to the top of the script and `Set-PSDebug -Off` at the end of the file; That is like `set -x` in bash; it would show you what is getting executed so you can debug it. Feel free to submit your fix in gist once you are done

**Comment by Slach** [comment]
Script works for me, thank @da-moon, maybe you could prepare PR into 
https://github.com/zed-industries/zed/tree/main/.github/workflows?

**Comment by vxxvvxxv** [comment]
I have an error when I'm trying to build:

```
   Compiling windows-core v0.58.0
   Compiling imagesize v0.12.0
   Compiling byteorder-lite v0.1.0
   Compiling slice-group-by v0.3.1
   Compiling lebe v0.5.2
   Compiling pico-args v0.5.0
   Compiling roxmltree v0.19.0
   Compiling leb128 v0.2.5
error: failed to run custom build command for `tree-sitter v0.22.6 (https://github.com/tree-sitter/tree-sitter?rev=7f4a57817d58a2f134fe863674acad6bbf007228#7f4a5781)`

Caused by:
  process didn't exit successfully: `C:\Users\user\zed-build\target\release\build\tree-sitter-f0b12d5419d1f050\build-script-build` (exit code: 1)
  --- stderr


  error occurred: Command "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MT" "-O2" "-Z7" "-Brepro" "-I" "C:\\Users\\user\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\wasmtime-c-api-impl-21.0.1/include" "-I" "C:\\Users\\user\\.cargo\\git\\checkouts\\tree-sitter-60edc5fae03f2366\\7f4a578\\lib\\src" "-I" "C:\\Users\\user\\.cargo\\git\\checkouts\\tree-sitter-60edc5fae03f2366\\7f4a578\\lib\\src\\wasm" "-I" "C:\\Users\\user\\.cargo\\git\\checkouts\\tree-sitter-60edc5fae03f2366\\7f4a578\\lib\\include" "-W4" "-DTREE_SITTER_FEATURE_WASM=" "-Dstatic_assert(...)=" "-FoC:\\Users\\user\\zed-build\\target\\x86_64-pc-windows-msvc\\release\\build\\tree-sitter-ff47546e589aca9f\\out\\ca7a7379f2970ef2-lib.o" "-c" "C:\\Users\\user\\.cargo\\git\\checkouts\\tree-sitter-60edc5fae03f2366\\7f4a578\\lib\\src\\lib.c" with args cl.exe did not execute successfully (status code exit code: 2).


warning: build failed, waiting for other jobs to finish...
Copy-Item: C:\Users\user\Downloads\build-zed.ps1:164
Line |
 164 |  Copy-Item "target\$arch\release\zed.exe" -Destination $InstallDir
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path 'C:\Users\user\zed-build\target\x86_64-pc-windows-msvc\release\zed.exe' because it does not
     | exist.
```

**Comment by Slach** [comment]
@vxxvvxxv 
install C++ visual studio 
https://visualstudio.microsoft.com/visual-cpp-build-tools/
and windows 10 SDK
https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/
or windows 11 SDK
https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/

**Comment by ggottemo** [comment]
Doesn't work with latest versions of Powershell Core for me due to depreciated Get-WMIObject. Running it the old Windows Powershell seems to work and is an easy way around this though - in case anyone else has this issue.

**Reply by da-moon** [reply]
In all fairness, I really did not have to use the `Get-WMIObject` cmdlet; I just kept it for myself so I remember how I can use `Get-WMIObject` to get system architecture with `Get-WMIObject`  :)

I was working on an update to this script. I will remove It in the update once it is ready

**Comment by JackMatanky** [comment]
First off, thank you for the time and effort you have put into writing this script🙏

I went through the whole download process and through your's and others' comments managed to complete the download, but now, neither the desktop shortcut nor the zed.exe file manage to open the program. I went through all the troubleshooting steps and the only error message I got was when I run `zed` by itself:
```
C:\Users\Username\AppData\Local\Microsoft\WindowsApps\zed.ps1 : Failed to start Zed: Cannot validate argument on parameter 'ArgumentList'. The argument is null, empty, or an element of the argument collection contains a null value.
Supply a collection that does not contain any null values and then try the command again.
At line:1 char:1
+ zed
+ ~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,zed.ps1
```

I tried my best to see if I could find a solution through the comments here, searching the internet, and copilot/chatgpt, but after a few hours, it is not looking any better. If you have any advice as to what I can do to fix the issue and open the program, I would be very appreciative.

Thanks!

**Comment by yordyi** [comment]
Good job!

**Comment by ndunnett** [comment]
For anyone else who gets the dreaded "path too long" error despite having [long paths](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation) enabled:

```
Building Zed... This may take a while.
    Updating crates.io index
error: failed to get `pet` as a dependency of package `languages v0.1.0 (C:\Users\ndunnett.workstation\zed-build\crates\languages)`

Caused by:
  failed to load source for dependency `pet`

Caused by:
  Unable to update https://github.com/microsoft/python-environment-tools.git?rev=ffcbf3f28c46633abd5448a52b1f396c322e0d6c#ffcbf3f2

Caused by:
  path too long: 'C:/Users/ndunnett.workstation/.cargo/git/checkouts/python-environment-tools-903993894b37a7d2/ffcbf3f/crates/pet-conda/tests/unix/conda_env_without_manager_but_found_in_history/some_other_location/conda_install/conda-meta/python-fastjsonschema-2.16.2-py310hca03da5_0.json'; class=Filesystem (30)
```

This can be resolved by setting long paths in git, ie. `git config --global core.longpaths true`. See https://github.com/rust-lang/cargo/issues/13020.

**Reply by Sergio-opensource** [reply]
may you have happiness in your life 😊

**Comment by ndunnett** [comment]
Thanks for the script, by the way. One thing that has changed since you wrote this is that CMake is also a dependency now, but the build script still works fine.

**Comment by GuoJikun** [comment]
```bash
Invoke-RestMethod : {"message":"API rate limit exceeded for 178.157.58.21. (But here's the good news: Authenticated req
uests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github
.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
所在位置 D:\build-zed.ps1:19 字符: 17
+     $releases = Invoke-RestMethod -Uri $releasesUrl -Method Get
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod]，WebExce
    ption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```
![image](https://github.com/user-attachments/assets/aac10ca4-8f5e-4189-9eb9-8ecdaec208ea)

**Comment by yuzisee** [comment]
In case it's helpful for anyone else:

1. I use chocolatey for a lot of things, but in this case the https://community.chocolatey.org/packages/visualstudio2022buildtools suggested above during https://github.com/zed-industries/zed/discussions/16811#discussioncomment-10553648 appeared to be insufficient (it presumably doesn't select "Desktop development with C++" automatically, so you're missing the MSVC `cl.exe` and the rest of the C++ compiler).
   - **→ HOWEVER**, `choco install visualstudio-installer` allowed me to select "Desktop development with C++" which includes also Windows SDK for Windows 10/11 **and that worked fine. ←**
   - (I did have to manually add '_MSVC v… - VS 2022 C++ x64/x86 Specret-mitigated libs (Latest)_' as well, though.)
1. The normal administrator PowerShell didn't work for me — I needed to open "Developer PowerShell for VS 2022" as administrator (specifically the x86_64 version calling through `C:\Windows\System32\WindowsPowerShell`; not the 32-bit x86 version calling through `C:\Windows\SysWOW64\WindowsPowerShell`).
   - As long as the powershell command `[Environment]::Is64BitProcess` prints `true` everything worked as expected
1. I needed to run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope Process` before executing the script (according to https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_execution_policies at least)

WIth those three things, it installed just fine 
![image](https://github.com/user-attachments/assets/4ce3397c-7fe8-4da5-a7e5-bb5fa7277ad9)
 v0.161.2

**Comment by Ukrainis** [comment]
Found this script, decided to check. During install, had several issues, which was fixing on-the-run.
1. I have Git installation, only for Git bash, so, my Git wasn't in Path, so I needed to add it, to continue with script.
2. Detected architecture: x86_64-pc-windows-msvc
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
Building Zed... This may take a while.
error: could not find `Cargo.toml` in `C:\Users\user\zed-build` or any parent directory
Copy-Item : Cannot find path 'C:\Users\user\zed-build\target\x86_64-pc-windows-msvc\release\zed.exe' because it does not exist.
At C:\work\zed_build\build-zed.ps1:164 char:1
+ Copy-Item "target\$arch\release\zed.exe" -Destination $InstallDir
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\valent...release\zed.exe:String) [Copy-Item], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand

To fix 2-nd issue, had changed build directory in script to the one, where I already have cloned repository.

Some suggestions maybe:
1. Check, if Git present in Path, if not, propose user to enter it's destination.

**Comment by jswetzen** [comment]
I'm trying to install but it fails when building `livekit_server`. It seems I'm having the same error as [this issue](https://github.com/tokio-rs/prost/issues/1193) but I have no idea how to install protobuf-compiler and libprotoc-dev on Windows.

**Comment by ESHARK22** [comment]
How did ssh work here, since zed (ssh) remote does not have a windows implementation?

**Comment by da-moon** [comment]
There are a bunch of more dependencies . I will add some comments and update the gist with my comments 



@jswetzen i have not tried building live kit so I don't have a good answer for you now

@ESHARK22 it was for using windows host to ssh into a container or WSL . Zed does not have an extension like VSCode's WSL remote development or Docker remote Development and I pretty much need to work in Linux environment.
The challenge that I faced back when I was experimenting was that Zed's LLM integration does not work in remote environment which is somewhat of a deal breaker to me as I would rather use an editor like Helix if I can't use the LLM integration

**Comment by shadeov** [comment]
# Solving some problems:

## Fix Error build `aws-lc-sys`

```powershell
error: failed to run custom build command for `aws-lc-sys v0.25.0`
```

Fix solution:
```powershell
winget install nasm
```

## Fix Error build `wasmtime-c-api-impl`

```powershell
error: failed to run custom build command for `wasmtime-c-api-impl v24.0.2`
```

Fix solution:
```powershell
winget install cmake
```

## Fix Error build `msvc_spectre_libs`

```powershell
error: failed to run custom build command for `msvc_spectre_libs v0.1.2`
```

Fix solution:
```powershell
winget install -e --id Microsoft.VisualStudio.2022.BuildTools --override "--passive --wait --add Microsoft.VisualStudio.Component.VC.Runtimes.x86.x64.Spectre"
```

**Comment by yuzisee** [comment]
Anyone getting
```
error: linking with `link.exe` failed: exit code: 1120
⋮
          libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp_strtol referenced in function pkey_rsa_ctrl_str
          libaws_lc_sys-e337c567e8e7db31.rlib(p_dh.obj) : error LNK2001: unresolved external symbol __imp_strtol
          libaws_lc_sys-e337c567e8e7db31.rlib(p_dsa.obj) : error LNK2001: unresolved external symbol __imp_strtol
          libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp_getenv referenced in function aws_lc_0_25_0_OPENSSL_cpuid_setup
          libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vsscanf referenced in function sscanf
          libaws_lc_sys-e337c567e8e7db31.rlib(mem.obj) : error LNK2019: unresolved external symbol __imp_realloc referenced in function aws_lc_0_25_0_OPENSSL_vasprintf_internal
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_fputs referenced in function print_errors_to_file
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_bsearch referenced in function err_string_lookup
          libaws_lc_sys-e337c567e8e7db31.rlib(obj.obj) : error LNK2001: unresolved external symbol __imp_bsearch
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_strerror referenced in function err_reason_error_string
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_strdup referenced in function aws_lc_0_25_0_ERR_set_error_data
          OLDNAMES.lib(strdup.obi) : error LNK2001: unresolved external symbol __imp_strdup
          libaws_lc_sys-e337c567e8e7db31.rlib(stack.obj) : error LNK2019: unresolved external symbol __imp_qsort_s referenced in function aws_lc_0_25_0_OPENSSL_sk_sort
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_feof referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_ferror referenced in function file_read
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fgets referenced in function file_gets
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fopen referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fread referenced in function file_read
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fseek referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_ftell referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fwrite referenced in function file_write
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp__setmode referenced in function file_ctrl
          OLDNAMES.lib(strdup.obi) : error LNK2001: unresolved external symbol __imp__strdup         libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp_strtol referenced in function pkey_rsa_ctrl_str
          libaws_lc_sys-e337c567e8e7db31.rlib(p_dh.obj) : error LNK2001: unresolved external symbol __imp_strtol
          libaws_lc_sys-e337c567e8e7db31.rlib(p_dsa.obj) : error LNK2001: unresolved external symbol __imp_strtol
          libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp_getenv referenced in function aws_lc_0_25_0_OPENSSL_cpuid_setup
          libaws_lc_sys-e337c567e8e7db31.rlib(bcm.obj) : error LNK2019: unresolved external symbol __imp___stdio_common_vsscanf referenced in function sscanf
          libaws_lc_sys-e337c567e8e7db31.rlib(mem.obj) : error LNK2019: unresolved external symbol __imp_realloc referenced in function aws_lc_0_25_0_OPENSSL_vasprintf_internal
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_fputs referenced in function print_errors_to_file
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_bsearch referenced in function err_string_lookup
          libaws_lc_sys-e337c567e8e7db31.rlib(obj.obj) : error LNK2001: unresolved external symbol __imp_bsearch
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_strerror referenced in function err_reason_error_string
          libaws_lc_sys-e337c567e8e7db31.rlib(err.obj) : error LNK2019: unresolved external symbol __imp_strdup referenced in function aws_lc_0_25_0_ERR_set_error_data
          OLDNAMES.lib(strdup.obi) : error LNK2001: unresolved external symbol __imp_strdup
          libaws_lc_sys-e337c567e8e7db31.rlib(stack.obj) : error LNK2019: unresolved external symbol __imp_qsort_s referenced in function aws_lc_0_25_0_OPENSSL_sk_sort
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_feof referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_ferror referenced in function file_read
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fgets referenced in function file_gets
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fopen referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fread referenced in function file_read
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fseek referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_ftell referenced in function file_ctrl
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp_fwrite referenced in function file_write
          libaws_lc_sys-e337c567e8e7db31.rlib(file.obj) : error LNK2019: unresolved external symbol __imp__setmode referenced in function file_ctrl
          OLDNAMES.lib(strdup.obi) : error LNK2001: unresolved external symbol __imp__strdup
⋮
error: could not compile `rustls-platform-verifier` (lib) due to 1 previous error
```

I tried:
1. Open VIsual Studio Installer and click [Update]
2. Reboot computer
3. Remove C:\Users\…\AppData\Local\Programs\Zed\version.txt
4. Remove C:\Users\…\zed-build

---

### Update:
I think the following worked?

- Open **C:\Users\…\zed-build\Cargo.toml**
- Look for the `[profile.release]` section
- Change `debug = "limited"` to `debug = false`

![image](https://github.com/user-attachments/assets/c43a2e7e-37a0-482f-917d-075b9d0f998a)
